types/user-types.d.ts

```diff
@@ -1,5 +1,5 @@
 interface IUser {
-  id: number
+  id?: number // make it optional here is not the best idea, but...
```

Comment:
You can create 2 different interfaces, one with is as optional and another one as required and use them in the places you need, you can use utility types to generate a new type from an existing type so you don't have to type all preparties

Reply:
Done. 
type UserWithOptionalId = Partial<Pick<IUser, 'id'>> & Omit<IUser, 'id'>;